### PR TITLE
Add dummysigner in contrib/tools

### LINKED ISSUE #152: refac revocation txs signature panel
the process of signing the emergency tx, unvault emergency tx and cancel tx is not clear both in the code and for the user.
The user do not understand what to do and tabs are not clickable.

to fix this:
- list revocation txs with a disclaimer "sign this three txs"
- each revocation tx has its own state with a SignState attached

### LINKED ISSUE #162: dummy signer
Emulate a hardware wallet

## Usage

take xpriv (multiple xpriv ?) as arg
Example using the password manager `pass`:

```
cargo run -- $(pass my-revault-setup/xpriv)
```

## Communication

Emulator embed a TCP server.

## Inspiration

- [Ledger Speculos](https://developers.ledger.com/docs/speculos/start-here/) 
- [Specter simulator](https://github.com/cryptoadvance/specter-diy/blob/master/docs/simulator.md)

### LINKED ISSUE #166: HW: specify the interface
create a doc/HW.md

### PR DESCRIPTION
**Pull request by edouardparis**
close #152 
close #166 
close #162

### CODE REVIEW

**Comment by danielabrozzoni** [comment]
Great job! 😍 
Just one thing, I think this "Retry" button doesn't work:
Revault GUI_092

Also, I think in the future we should display at least the spend tx outputs, so the user can check if the destination addresses are correct.

The UI/UX is so cool now, I'm really happy with it!

**Comment by edouardparis** [comment]
Thank you !

>Also, I think in the future we should display at least the spend tx outputs, so the user can check if the destination addresses are correct.

Yes I didnt want to have a too big PRs

File: contrib/tools/dummysigner/README.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+{"request_status": "refused"}
+```
+
+### Sign spend transaction
+
+#### request:
+
+```json
+{
+  "derivation_path": "<string ex: m/1>",
```
Comment (edouardparis) [inline contrib/tools/dummysigner/README.md:42]: my bad documentation is not good here
```suggestion
  "derivation_paths": ["<string ex: m/1>"], // paths ordered by respective spend tx pbst input index
```

File: contrib/tools/dummysigner/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+[package]
+name = "dummysigner"
+version = "0.1.0"
+edition = "2018"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+revault_tx = { git = "https://github.com/revault/revault_tx", features = ["use-serde"] }
```
Comment (darosior) [inline contrib/tools/dummysigner/Cargo.toml:9]: Prefer a release to not get broken. `0.0.3`

File: contrib/tools/dummysigner/src/sign.rs [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    pub fn serialize<'se, S>(
+        psbt: &PartiallySignedTransaction,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&base64::encode(encode::serialize(&psbt)))
+    }
+}
```
Comment (darosior) [inline contrib/tools/dummysigner/src/sign.rs:80]: You know that we have code that does that in revault_tx, right? :sweat_smile:

**Review by darosior — COMMENTED** [review]
Some comments (don't bother addressing the one about revault_tx), will test now

File: contrib/tools/dummysigner/src/sign.rs (hunk 3 above)
Comment (edouardparis) [reply contrib/tools/dummysigner/src/sign.rs:80]: I am not sure where it is. I would still prefer to keep this code and not rely on an external dep.

**Review by danielabrozzoni — APPROVED** [review]
ACK 5d462fee59ccc25ded6d21141b7db6d387f79db7

**Pull request merged**